hail/python/hailtop/aiogoogle/client/storage_client.py

```diff
@@ -254,12 +254,12 @@ async def listfiles(self, url: str, recursive: bool = False) -> AsyncIterator[Fi
 
         it = it.__aiter__()
         try:
-            n = await it.__anext__()
+            first_entry = await it.__anext__()
         except StopAsyncIteration:
             raise FileNotFoundError(url)  # pylint: disable=raise-missing-from
 
         async def cons(n, it):
-            yield n
+            yield first_entry
```

Comment:
This isn't quite right. The cons takes an argument that should be first_entry. Can you rename `n` to `head` and it's invoked `cons(first_entry, it)`?

Reply:
Sorry about that, fixed.